lib/DxcSupport/WinFunctions.cpp

```diff
@@ -86,6 +104,35 @@ int _stricmp(const char *str1, const char *str2) {
   return str1[i] - str2[i];
 }
 
+int _wcsicmp(const wchar_t *str1, const wchar_t *str2) {
+  size_t i = 0;
+  for (; str1[i] && str2[i]; ++i) {
+    int d = std::towlower(str1[i]) - std::towlower(str2[i]);
+    if (d != 0)
+      return d;
+  }
+  return str1[i] - str2[i];
+}
+
+int _wcsnicmp(const wchar_t *str1, const wchar_t *str2, size_t n) {
```

Comment:
Is the following statement true?
> _wcsnicmp should only access and compare the first n elements of str1 and str2

if so, your implementation is incorrect in that the for loop could access more than n elements.
I suggest using
```cpp
for(; i < n && str[i] && str2[i]; ++i)
```
Is this correct or am I missing something?

Reply:
Yeah. Looks like I botched an earlier correction.